p2p/transport/webrtc/errors.go

```diff
@@ -0,0 +1,49 @@
+package libp2pwebrtc
+
+import (
+	"fmt"
+)
+
+type errKind string
+
+const (
+	errKindConnectionFailed errKind = "peerconnection failed"
+	errKindDatachannel      errKind = "datachannel"
+	errKindMultiaddr        errKind = "bad multiaddr"
+	errKindNoise            errKind = "noise"
+	errKindInternal         errKind = "internal"
+)
+
+var (
+	errDataChannelTimeout = errDatachannel("timed out waiting for datachannel", nil)
+)
+
+type webRTCTransportError struct {
+	kind    errKind
+	message string
+	nested  error
```

Comment:
This is borderline incorrect go code as you are not implemeting `Unwrap` with nested.

You should add a:
```go
func (e webRTCTransportError) Unwrap() error {
  return e.nested
}```